### DISCUSSION TITLE
Poor Fit to Seemingly Simple Models?

### DISCUSSION THREAD

**Discussion by benne94** [pastas/pastas#669] [Q&A] [answered]
Hi all,

I am currently trialling PASTAS and have still not used the library a great deal so looking for some help.

I have run PASTAS models for seemingly simple time-series of groundwater levels that show strong influence of rainfall.

I've fit the model using several configurations:

1. Using rainfall only
2. Using rainfall - evap
3. Using a Recharge model using rainfall and evap
4. Configuring the above with various response functions
5. Standardising the groundwater level time-series
6. Changing frequency from 1D, 3D, 1W, etc.

I'm struggling to get a result better than the screenshot below and was wondering if anyone could offer any help.  I've tried changing the warmup period too but can't seem to get a good fit.  I notice in the example material time-series generally cover 10-20 years of data and was thinking perhaps my data coverage isn't good enough.  I was therefore thinking I could copy the time-series backwards so that I have a longer period for the model to calibrate.

ml = ps.Model(oseries=df_vwp_filt_rs_s, name=s)

rch = ps.rch.FlexModel(gw_uptake=True)
sm = ps.RechargeModel(df_rain, 
                      df_evp, 
                      recharge=rch, 
                      rfunc=ps.Exponential(), 
                      name='Recharge')

ml.add_stressmodel(sm)

ml.solve(#tmin='2019-01-01', 
         #tmax='2023-12-01', 
         warmup=450,
         #solver=ps.LmfitSolve()
         )
ml.plots.results(figsize=(12,8))

**Results**
![image](https://github.com/pastas/pastas/assets/83494177/5594af34-e3a7-47ba-9bc7-4b7a9932d838)
**Time Series**
![image](https://github.com/pastas/pastas/assets/83494177/6aee538f-e976-4118-8c86-318635f0f04a)

**Comment by mbakker7** [comment]
First thing to try: Fit model without a noise model:

`ml.solve(noise=False)`

**Reply by AviadAvr** [reply]
Hi Mark, 
I was running into a similar problem and this seemed to work. 
My problem is with calculating a model for high-frequency observations (hourly, to be exact, for a year or so) and I never seem to get a good fit with the noise model.
To my understanding, the noise model is used to un-correlate the residuals from themselves by converting them into, well... noise.

Is it correct to assume that, for higher-frequency data, the noise model can be ignored because, due to the frequency of sampling, the data is noisy by itself?
In that case, what would be a rule of thumb for deciding when to ignore the noise model? should i check if my measurement noise follows a gausian distribution and decide based on that?

Thanks

**Reply by mbakker7** [reply]
The noise model that is currently implemented in Pastas is a fairly simply AR1 model. It is probably the simplest noise model you can think off. It works pretty well for a lot of cases, but, unfortunately, there are also quite a few cases where it is not sufficient. In that respect, it is better to start your modeling without a noise model. When you are interested in uncertainties, then you have to check your residuals for autocorrelation and, if there is significant autocorrelation, try to see if the AR1 noise model is able to resolve the autocorrelation. Our experience is that the AR1 noise model works less good for data with a higher frequency. 

To promote this approach of modeling (starting without a noise model and only add it when it is needed), a future version of Pastas (probably starting with version 2.0) will have the noise model turned off as the default and leave it to the modeler to add it when it is needed.

**Comment by raoulcollenteur** [comment]
I think Marks advice should improve things. You might also try `gw_uptake=False` (this is only an experimental feature and not recommended) and make sure precipitation and evaporation are in mm/d for the nonlinear model.

**Reply by mbakker7** [reply]
Just out now curiosity, what does `gw=False` try to do?

**Reply by raoulcollenteur** [reply]
My bad, should have been `gw_uptake=False`. It just means there's no evaporation from the groundwater...

**Comment by benne94** [comment]
Thanks for the quick replies.  The units are in mm/d for the rainfall and evaporation and groundwater levels are daily in masl.

I ran with noise model = False (image 1) and with GW uptake off (image 2) as well as with rainfall and evap as seperate stress models (image 3).

Image 1:

![image](https://github.com/pastas/pastas/assets/83494177/76defb63-d066-4a6d-a0b4-f01989769111)


Image 2: 
![image](https://github.com/pastas/pastas/assets/83494177/abd27e3d-1f90-43db-8037-866235320c38)

Image 3: 
![image](https://github.com/pastas/pastas/assets/83494177/d9b93181-be40-426b-a81e-6d5b438b7751)

The files I am using are here and evaporation is from monthly daily averages:

[df_rain.csv](https://github.com/pastas/pastas/files/13510396/df_rain.csv)
[df_gwls.csv](https://github.com/pastas/pastas/files/13510395/df_gwls.csv)
[df_evap.csv](https://github.com/pastas/pastas/files/13510394/df_evap.csv)

**Comment by raoulcollenteur** [comment]
I had a quick Look at the time series. Are you sure the rain data is correct? E.g., it rains a lot when the GWL is dropping? In that case I guess there's a thick unsaturated zone and a large delay in the head response, so you might try the FourParam response function :) 

![image](https://github.com/pastas/pastas/assets/12051167/8655387b-dd26-4dfa-9963-bc4ddb6d1e0b)

**Reply by benne94** [reply]
Thanks Raoul,  

I have tried this now, you're correct there's a lag between rainfall and GWLs of several months.  I tried FourParams but still not getting great results.  

Have you had success with PASTAS on this short of a time-series?

![image](https://github.com/pastas/pastas/assets/83494177/bbf915bd-f760-4d69-acac-6bea61496bc5)

Thanks,
Adam

**Reply by raoulcollenteur** [reply]
Hi Adam,

Looks like a difficult head series to model! Normally you would still get a decent fit with shorter time series. I think its rather the data here. Have you tried any other model for this data? I would be interested to see if other approaches might work. Though when looking at the input data and the heads I can't imagine other models will work much better but it's worth trying. Perhaps there's a river nearby?

Cheers,
Raoul

**Reply by benne94** [reply]
Hi Raoul,

I have done a trial of using other ML methods based on some of the available literatures (using CNNs and LSTM models).  What I really like with Pastas is it feels much more explainable with the well documented response functions which are influenced by our physical understanding of groundwater systems, as opposed to the typically 'black box' nature of ML models.

I'd be interested to know of any other approaches you would consider for these problems as I'm still in the early stages of looking at this area.

Many thanks,
Adam

**Reply by mbakker7** [reply, marked as answer]
Adam,
You may want to check out the paper Solving Groundwater Flow Problems with Time Series Analysis: You May Not Even Need Another Model https://ngwa.onlinelibrary.wiley.com/doi/full/10.1111/gwat.12927, which discusses exactly what you mean: lumped-parameter models with response functions (like Pastas) are often easier to understand. Hope this helps.

**Reply by benne94** [reply]
Thanks Mark,  I will give this a read and let you know if I find anything that works.  I will mark your response as the answer as I think you've given plenty of options.

Thanks,
Adam
